src/main/java/seedu/nuke/command/promptcommand/DeleteConfirmationPrompt.java

```diff
@@ -0,0 +1,370 @@
+     */
+    private CommandResult executeMultipleDelete(ArrayList<Directory> filteredList) {
+        ArrayList<Integer> toDeleteIndices = Executor.getIndices();
+        switch (directoryLevel) {
+
+        case MODULE: {
+            // Cast to Array List of modules
+            ArrayList<Module> filteredModules = filteredList.stream()
+                    .map(Module.class::cast)
+                    .collect(Collectors.toCollection(ArrayList::new));
+            deleteMultipleModules(filteredModules, toDeleteIndices);
+            return new CommandResult(MESSAGE_DELETE_MODULE_SUCCESS);
+        }
+
+        case CATEGORY: {
+            // Cast to Array List of categories
+            ArrayList<Category> filteredCategories = filteredList.stream()
+                    .map(Category.class::cast)
+                    .collect(Collectors.toCollection(ArrayList::new));
+            deleteMultipleCategories(filteredCategories, toDeleteIndices);
+            return new CommandResult(MESSAGE_DELETE_CATEGORY_SUCCESS);
+        }
+
+        case TASK: {
+            // Cast to Array List of tasks
+            ArrayList<Task> filteredTasks = filteredList.stream()
+                    .map(Task.class::cast)
+                    .collect(Collectors.toCollection(ArrayList::new));
+            deleteMultipleTasks(filteredTasks, toDeleteIndices);
+            return new CommandResult(MESSAGE_DELETE_TASK_SUCCESS);
+        }
+
+        case FILE: {
+            // Cast to Array List of files
+            ArrayList<TaskFile> filteredFiles = filteredList.stream()
+                    .map(TaskFile.class::cast)
+                    .collect(Collectors.toCollection(ArrayList::new));
+            try {
+                deleteMultipleFiles(filteredFiles, toDeleteIndices);
+                return new CommandResult(MESSAGE_DELETE_FILE_SUCCESS);
+            } catch (FileNotFoundException e) {
+                return new CommandResult(MESSAGE_FILE_NOT_FOUND);
+            } catch (FileSystemException e) {
+                return new CommandResult(MESSAGE_FILE_SYSTEM_EXCEPTION);
+            } catch (IOException e) {
+                return new CommandResult(MESSAGE_DELETE_FILE_ERROR);
+            } catch (SecurityException e) {
+                return new CommandResult(MESSAGE_FILE_SECURITY_EXCEPTION);
+            }
+        }
+
+        default:
+            return new CommandResult("Error in multiple deletion.");
+        }
+    }
+
+    /**
+     * Checks if any of the directories to be deleted is the current directory or its parent.
+     *
+     * @param filteredList
+     *  The filtered list of directories to be deleted
+     * @return
+     *  <code>TRUE</code> if the current directory or its parent is being deleted, or <code>FALSE</code> otherwise
+     * @throws IncorrectDirectoryLevelException
+     *  If there is illegal access to a directory
+     */
+    private boolean isCurrentDirectoryInList(ArrayList<Directory> filteredList)
```

Comment:
![image](https://user-images.githubusercontent.com/38585175/77996584-7712d900-7360-11ea-9abf-dd5a8e95402c.png)
Haven't been written in the code. Is it not correct?